h2/src/main/org/h2/command/Command.java

```diff
@@ -251,54 +255,57 @@ public ResultWithGeneratedKeys executeUpdate(Object generatedKeysRequest) {
                 // wait
             }
         }
-        synchronized (sync) {
-            Session.Savepoint rollback = session.setSavepoint();
-            session.setCurrentCommand(this, generatedKeysRequest);
-            try {
-                while (true) {
-                    database.checkPowerOff();
-                    try {
-                        int updateCount = update();
-                        if (!Boolean.FALSE.equals(generatedKeysRequest)) {
-                            return new ResultWithGeneratedKeys.WithKeys(updateCount,
-                                    session.getGeneratedKeys().getKeys(session));
-                        }
-                        return ResultWithGeneratedKeys.of(updateCount);
-                    } catch (DbException e) {
-                        start = filterConcurrentUpdate(e, start);
-                    } catch (OutOfMemoryError e) {
-                        callStop = false;
-                        database.shutdownImmediately();
-                        throw DbException.convert(e);
-                    } catch (Throwable e) {
-                        throw DbException.convert(e);
+        Session.Savepoint rollback = session.setSavepoint();
+        session.startStatementWithinTransaction();
+        session.setCurrentCommand(this, generatedKeysRequest);
```

Comment:
The same problem here.

Reply:
Yes, I think I should go back to what it was. Protection of session fields is a mess... :disappointed: